sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_pyamqp_transport.py

```diff
@@ -804,15 +807,26 @@ def settle_message_via_receiver_link(
             if settle_operation == MESSAGE_DEFER:
                 return handler.settle_messages(
                     message._delivery_id,
+                    message._delivery_tag,
                     'modified',
                     delivery_failed=True,
                     undeliverable_here=True
                 )
         except AttributeError as ae:
             raise RuntimeError("handler is not initialized and cannot complete the message") from ae
 
+        except AMQPLinkError as le:
+            if (
+                le.condition == ErrorCondition.InternalError
+                and isinstance(le.description, str)
+                and le.description.startswith("Delivery tag")
+            ):
+                raise RuntimeError("Link error occurred during settle operation.") from le
+
+            raise ServiceBusConnectionError(message="Link error occurred during settle operation.") from le
+
         except AMQPConnectionError as e:
-            raise RuntimeError("Connection lost during settle operation.") from e
+            raise ServiceBusConnectionError(message="Connection lost during settle operation.") from e
```

Comment:
is this a breaking change? If users were catching on RuntimeError before, it will no longer be caught.

Reply:
This error is still RuntimeError in the async code, so will need to be updated.

Reply:
Il need to look over this one but a `RuntimeError` should not bubble up to users at all.

Reply:
yeah this one is a mistake. Ill revert it back to RuntimeError